tests/src/test/scala/org/apache/openwhisk/core/limits/ActionLimitsTests.scala

```diff
@@ -463,4 +465,38 @@ class ActionLimitsTests extends TestHelpers with WskTestHelpers with WskActorSys
       _.response.result.get.fields("error") shouldBe Messages.memoryExhausted.toJson
     }
   }
+
+  /**
+   * Test that a heavy logging action is interrupted within its timeout limits.
+   */
+  it should "interrupt the heavy logging action within its time limits" in withAssetCleaner(wskprops) {
+    (wp, assetHelper) =>
+      val name = s"NodeJsTestLoggingActionCausingTimeout-${System.currentTimeMillis()}"
+      print(s"\n create action ${name} using api host: ${wskprops.apihost}..")
+      assetHelper.withCleaner(wsk.action, name, confirmDelete = true) { (action, _) =>
+        action.create(
+          name,
+          Some(TestUtils.getTestActionFilename("loggingTimeout.js")),
+          timeout = Some(allowedActionDuration))
+      }
+      val duration = allowedActionDuration + 3.minutes
+      val checkDuration = allowedActionDuration + 1.minutes
+      val run =
+        wsk.action.invoke(name, Map("durationMillis" -> duration.toMillis.toJson, "delayMillis" -> 100.toJson))
+      withActivation(wsk.activation, run) { result =>
+        withClue("Activation result not as expected:") {
+          result.response.status shouldBe ActivationResponse.messageForCode(ActivationResponse.DeveloperError)
+          result.response.result.get.fields("error") shouldBe {
+            Messages.timedoutActivation(allowedActionDuration, init = false).toJson
+          }
+          val logs = result.logs.get
+          logs.last should include(Messages.logFailure)
+
+          val parseLogTime = (line: String) => Instant.parse(line.split(' ').head)
+          val startTime = parseLogTime(logs.head)
+          val endTime = parseLogTime(logs.last)
+          between(startTime, endTime) should be < checkDuration.asInstanceOf[Duration]
```

Comment:
You could get rid of the type cast as follows:

```
val duration = between(startTime, endTime)
FiniteDuration(duration.length, duration.unit) should be < allowedActionDuration
```

I checked other usages of `TimingHelpers.between()` and `TimingHelpers.durationOf()`. Actually, it's only used in a few locations. And honestly, I don't understand why these two methods return a `Duration` instead of a `FiniteDuration`. You PR could just change to a `FiniteDuration` - that would make a lot of sense.

The other locations using `TimingHelpers.durationOf()` work around the problem you observed with comparing a `Duration` to a `FiniteDuration` by converting the durations to milliseconds and comparing the resulting values... See https://github.com/apache/incubator-openwhisk/blob/master/tests/src/test/scala/org/apache/openwhisk/core/containerpool/docker/test/DockerContainerTests.scala#L633.

Reply:
Good call. I wondered about this myself yesterday. Feel free to change the between implementation to FiniteDuration (it's guaranteed to be one anyway)

Reply:
We changed both methods in `TimingHelpers` to return results using `FiniteDuration` but unfortunately as already mentioned above it seems that we cannot simply compare two `FiniteDuration` values using `should be <` (so we moved back to millis, as said if anyone has an idea how solve please let us know).

Reply:
@markusthoemmes I think we should accept the latest results and proposal by @steffenrost @ruediger-maass and compare the `Duration`s millis instead of the `Duration`s millis. The Scala compile error from above is hard to solve and - apparently - nobody else solved it so far because the whole project seems to have no `Duration` comparisons with `should be`.

@mhenke1 found a solution where you need to define an implicit in scope to make sure that `should be` is able to compare `Duration`s properly. Not very comfortable to read...